src/main/java/uk/gov/hmcts/darts/arm/mapper/impl/MediaArchiveRecordMapperImpl.java

```diff
@@ -230,24 +233,33 @@ private String getHearingDate(MediaEntity media) {
 
     private String getCaseNumbers(MediaEntity media) {
         String cases = null;
-        if (nonNull(media.getHearingList())) {
-            List<HearingEntity> hearings = media.getHearingList();
-            List<String> caseNumbers = hearings
-                    .stream()
-                    .map(HearingEntity::getCourtCase)
-                    .map(CourtCaseEntity::getCaseNumber)
-                    .toList();
+        if (CollectionUtils.isNotEmpty(media.getHearingList())) {
+            List<String> caseNumbers = media.getHearingList()
+                .stream()
+                .map(HearingEntity::getCourtCase)
+                .map(CourtCaseEntity::getCaseNumber)
+                .toList();
             cases = caseListToString(caseNumbers);
         }
         return cases;
     }
 
     private static String getCourthouse(MediaEntity media) {
```

Comment:
This probably needs @hemantasharma1129 to confirm, but I think if there is a preference for us to go through the links to Hearing from Annotation, Media, Transcription then I think the getCourthouse and getCourtroom should be refactored to do that using a`hearing` method param?  It's just not clear to me how we are expected to handle the migrated records as I thought that was why we also end up having courtroom links from transcription say which is nullable.  Yet we have courtroom link from Media that is mandatory currently.

I recall having to do DMP-1569 Change MediaId Authorisation to use Hearing (#942) recently, so got to say I'm still left confused about these database links and their nullability which is causing us issues in the code.

Reply:
[DMP-2410](https://tools.hmcts.net/jira/browse/DMP-2410) ARM Metadata - Verify the metadata population of fields